app/src/main/java/com/example/henzoshimada/feeltrip/GPSLocation.java

```diff
@@ -0,0 +1,215 @@
+package com.example.henzoshimada.feeltrip;
+
+import android.app.Activity;
+import android.app.Service;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.os.IBinder;
+import android.provider.Settings;
+import android.support.v4.app.ActivityCompat;
+import android.support.v7.app.AlertDialog;
+import android.util.Log;
+
+import static android.content.Context.LOCATION_SERVICE;
+
+/**
+ * Created by Kevin on 16-Mar-17.
+ */
+//http://stackoverflow.com/questions/1513485/how-do-i-get-the-current-gps-location-programmatically-in-android
+public class GPSLocation extends Service implements LocationListener {
+
+    private final Context mContext;
+
+    // Flag for GPS status
+    boolean isGPSEnabled = false;
+
+    // Flag for network status
+    boolean isNetworkEnabled = false;
+
+    // Flag for GPS status
+    boolean canGetLocation = false;
+
+    Location location; // Location
+    double latitude; // Latitude
+    double longitude; // Longitude
+
+    // The minimum distance to change Updates in meters
+    private static final long MIN_DISTANCE_CHANGE_FOR_UPDATES = 10; // 10 meters
+
+    // The minimum time between updates in milliseconds
+    private static final long MIN_TIME_BW_UPDATES = 1000 * 60 * 1; // 1 minute
```

Comment:
This file is found online, these two constants are used to decide the minimum change of distance for the location to change and the minimum time between updating the location